main.tf

```diff
@@ -42,56 +41,76 @@ module "project-services" {
     "sql-component.googleapis.com",
     "sqladmin.googleapis.com",
     "storage.googleapis.com",
-    "secretmanager.googleapis.com",
     "run.googleapis.com",
     "redis.googleapis.com",
   ]
 }
 
 resource "google_service_account" "runsa" {
   project      = var.project_id
-  account_id   = "${var.deployment_name}-run-sa"
+  account_id   = var.deployment_name
   display_name = "Service Account for Cloud Run"
 }
 
 resource "google_project_iam_member" "allrun" {
-  project    = data.google_project.project.number
-  role       = "roles/secretmanager.secretAccessor"
-  member     = "serviceAccount:${google_service_account.runsa.email}"
-  depends_on = [module.project-services]
+  for_each = toset(var.run_roles_list)
+  project  = data.google_project.project.number
+  role     = each.key
+  member   = "serviceAccount:${google_service_account.runsa.email}"
 }
 
-module "network-safer-mysql-simple" {
-  source  = "terraform-google-modules/network/google"
-  version = "~> 4.0"
+resource "random_id" "name" {
+  byte_length = 2
+}
 
-  project_id   = var.project_id
-  network_name = "${var.deployment_name}-network"
+resource "google_compute_network" "main" {
+  provider                = google-beta
+  name                    = "${var.deployment_name}-private-network"
+  auto_create_subnetworks = true
+  project                 = var.project_id
+}
 
-  subnets = []
-  depends_on = [
-    module.project-services
-  ]
+resource "google_compute_global_address" "main" {
+  name          = "${var.deployment_name}-vpc-address"
+  provider      = google-beta
+  purpose       = "VPC_PEERING"
+  address_type  = "INTERNAL"
+  prefix_length = 16
+  network       = google_compute_network.main.name
+  project       = var.project_id
 }
 
-module "private-service-access" {
-  source      = "GoogleCloudPlatform/sql-db/google//modules/private_service_access"
```

Comment:
Any reason for switching the modules to resources?

Reply:
1. Was using a module for setting up networks that was tailored for  mysql, not postgres, so I should replace that anyway. 
2. I wasn't saving myself a lot of effort by using them. 
3. I was troubleshooting the issue with destroy not working, and needed granular control over all of the parts to troubleshoot. (selectively commenting out bits of the infrastructure to see if they were leading to this not working)